jetty-websocket/websocket-servlet/src/main/java/org/eclipse/jetty/websocket/servlet/WebSocketUpgradeFilter.java

```diff
@@ -123,6 +125,19 @@ public static FilterHolder ensureFilter(ServletContext servletContext)
             servletHandler.prependFilter(holder);
             servletHandler.prependFilterMapping(mapping);
 
+            // If we create the filter we must also make sure it is removed if the context is stopped.
+            contextHandler.addEventListener(new LifeCycle.Listener()
+            {
+                @Override
+                public void lifeCycleStopping(LifeCycle event)
+                {
+                    FilterHolder[] holders = Arrays.stream(servletHandler.getFilters())
+                        .filter(h -> h != holder)
+                        .toArray(FilterHolder[]::new);
+                    servletHandler.setFilters(holders);
+                }
+            });
```

Comment:
@sbordet not sure about this one. Seems like we should be removing the `WebSocketUpgradeFilter` if it was automatically created and not added manually.  But this doesn't seem like the right way to do it.

Reply:
Is there a mapping you need to remove as well?

Reply:
Good point, I will add this in.

Reply:
@lachlan-roberts I don't think you should remove it.
If the context is stopped, and then restarted, you first check if the filter is already there, so you won't add it twice.